src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,152 @@
+package duke.parser;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+import duke.duke.Duke;
+import duke.exceptions.InvalidArgumentException;
+import duke.exceptions.InvalidCommandException;
+import duke.utils.DateValidator;
+import duke.utils.DateValidatorUsingDateFormat;
+
+/**
+ * Parses user input.
+ */
+public class Parser {
+
+    /**
+     * Parses user input into a command.
+     * @param userInput User input string.
+     * @param bot A Duke object that manages task list operations.
+     * @return User command.
+     * @throws InvalidCommandException If user input is an unrecognised command.
+     * @throws InvalidArgumentException If argument is missing from user input.
+     */
+    public static Command processInput(String userInput, Duke bot) throws InvalidCommandException,
+            InvalidArgumentException {
+        String[] userInputArray = userInput.split(" ", 2);
+        Command userCommand = null;
+        if (userInputArray.length == 1) {
+            if (!(userInputArray[0].equals("bye") || userInputArray[0].equals("list"))) {
+                if (userInputArray[0].equals("todo") || userInputArray[0].equals("done")
+                        || userInputArray[0].equals("deadline") || userInputArray[0].equals("event")
+                        || userInputArray[0].equals("delete")) {
+                    throw new InvalidCommandException("OOPS!!! " + "The description of a "
+                            + userInputArray[0] + " cannot be empty.\n");
+                } else {
+                    throw new InvalidCommandException("OOPS!!! "
+                            + "I'm sorry, but I don't know what that means :-()\n");
+                }
+            }
+
+            if (userInputArray[0].equals("bye")) {
+                userCommand = new ByeCommand();
+            } else {
+                userCommand = new ListCommand();
+            }
+        } else {
+            switch (userInputArray[0]) {
+            case "done":
```

Comment:
It seems like your parsing for each command is pretty long. Could have broken down into functions to increase readability.